### ISSUE TITLE
RDF literal indexing thread dies

### ISSUE THREAD

**Issue by rla** [SWI-Prolog/packages-semweb#13]
RDF literal indexing gives warning and does not work:

?- run.
% Parsed "data.ttl" in 0.00 sec; 140 triples
Loaded
Warning: [Thread __rdf_tokenizer] Thread running "make_literal_index(<message_queue>(0x7feb98024600))" died on exception: double_metaphone:double_metaphone/2: Type error: `text' expected, found `0' (an integer)
false.

Test file (data.ttl) and test code are added in gist:
https://gist.github.com/rla/4c5311009e2766acf9fa

SWI-Prolog version is 7.3.7

**Comment by wouterbeek** [comment]
The problem is that `rdf_find_literal(sounds(advisory), Lit)` uses double metaphone, which does not work for non-text literals:

```prolog
?- double_metaphone(0, X).
ERROR: double_metaphone:double_metaphone/2: Type error: `text' expected, found `0' (an integer)
```

Maybe non-text literals should be excluded from the results of `sounds/1` altogether?

**Comment by JanWielemaker** [comment]
It tried to avoid that, except from one place.  Should be fixed with 7076bee36f187cc7148d3e97bf33630bc6a9cfa7
